### PR TITLE
Fix crash with r_vertexLighting

### PR DESCRIPTION
**Pull request by slipher**
Also show a useful error message when crashing from this class of bug (using an "unused" permutation).

### CODE REVIEW

File: src/engine/renderer/gl_shader.cpp [hunk 1]
```diff
@@ -1041,11 +1039,6 @@ void GLShaderManager::CompileGPUShaders( GLShader *shader, shaderProgram_t *prog
 			/* FIXME: add this test: ( strcmp( token, "USE_TCGEN_LIGHTMAP" ) == 0 && r_lightMapping->integer == 0 )
 			when lightmaps are never used when lightmapping is disabled
 			see https://github.com/DaemonEngine/Daemon/issues/296 is fixed */
-			if ( strcmp( token, "USE_LIGHT_MAPPING" ) == 0 && r_vertexLighting->integer != 0 )
```
Comment (illwieckz) [inline src/engine/renderer/gl_shader.cpp:1044]: The comment preceding this line is now meaningless (also `is fixed` was a leftover).

Maybe rewrite the comment saying we may want to not build it when surface isn't lightmapped or lightmapping is disabled but this currently does not work (with a link to the issue/PR)?

**Comment by illwieckz** [comment]
Do we know when light mapping code is used when light mapping is disabled?

Anyway I consider this a correct fix for now, as it allows us to release a non-crashing engine, but we may want to track the root issue of using the light mapping code when light mapping is disabled.

I now remind that maps using lightstyles (like metro) was previously still drawing lightmaps on lightstyled surfaces even when grid lighting was enabled: https://github.com/DaemonEngine/Daemon/issues/296

Maybe we still have other issues like that.

**Comment by slipher** [comment]
> Do we know when light mapping code is used when light mapping is disabled?

It's a shader with an ST_LIGHTMAP stage. 

In `Render_lightMapping` we have 
```
	bool enableLightMapping = !r_vertexLighting->integer \
		&& tess.bspSurface \
		&& tess.lightmapNum >= 0 && tess.lightmapNum <= tr.lightmaps.currentElements;
```

but then...

```
	if ( noLightMap )
	{
		lightmap = tr.whiteImage;
		enableLightMapping = true;
	}
```

File: src/engine/renderer/gl_shader.cpp (hunk 1 above)
Comment (slipher) [reply src/engine/renderer/gl_shader.cpp:1044]: I don't understand it well enough to write an intelligent comment myself.

**Comment by illwieckz** [comment]
Oh I see, If I'm right those are surfaces without lightmap stage!!! We lit them with a dummy lightmap that is a pure white image. Maybe we can do the same dummy lightgrid instead.

**Comment by illwieckz** [comment]
Can you add a comment like that in `tr_shade.cpp`?

```diff
+	/* FIXME: This requires light mapping GLSL code to be compiled even when light mapping is disabled.
+	See https://github.com/DaemonEngine/Daemon/issues/500 */
	if ( noLightMap )
	{
		lightmap = tr.whiteImage;
		enableLightMapping = true;
	}
```

**Comment by slipher** [comment]
Added the comment around the noLightMap hack and deleted the one by the unusedPermutation stuff.

**Comment by necessarily-equal** [comment]
The code looks okish to me, but then I don't know anything about that part of the code or that programming realm — so I guess you can count that as weak approval?

**Review by DolceTriade — APPROVED** [review]

**Review by illwieckz — APPROVED** [review]
LGTM.

**Pull request merged**